### PR TITLE
[#1737] Improvement(postgresql): Obtain PostgreSQL table information in the form of metadata using the driver 

### PR DESCRIPTION
**Pull request by Clearvive**
### What changes were proposed in this pull request?
Obtain PostgreSQL table information in the form of metadata using the driver 

### Why are the changes needed?
Fix: #1737

### Does this PR introduce _any_ user-facing change?
NO

### How was this patch tested?
UT/IT

### CODE REVIEW

File: catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/converter/JdbcTypeConverter.java [hunk 1]
```diff
@@ -5,16 +5,17 @@
 package com.datastrato.gravitino.catalog.jdbc.converter;
 
 import com.datastrato.gravitino.rel.types.Type;
+import java.util.Objects;
 
-public abstract class JdbcTypeConverter<FROM, TO> {
+public abstract class JdbcTypeConverter<TO> {
```
Comment (mchades) [inline catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/converter/JdbcTypeConverter.java:10]: Please add a comment for the generalization `TO`

File: catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/converter/JdbcTypeConverter.java [hunk 2]
```diff
@@ -5,16 +5,17 @@
...
 
   /**
    * Convert from Gravitino type to JDBC type
    *
    * @param type
    * @return
```
Comment (mchades) [inline catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/converter/JdbcTypeConverter.java:16]: Please fill the comment

File: catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/converter/JdbcTypeConverter.java [hunk 3]
```diff
@@ -23,4 +24,68 @@ public abstract class JdbcTypeConverter<FROM, TO> {
...
    */
   public abstract TO fromGravitinoType(Type type);
+
+  public static class JdbcTypeBean {
+    private String typeName;
+    private String columnSize;
+    private String scale;
```
Comment (mchades) [inline catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/converter/JdbcTypeConverter.java:31]: Please give some comments for those fields

File: catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/converter/JdbcTypeConverter.java [hunk 4]
```diff
@@ -23,4 +24,68 @@ public abstract class JdbcTypeConverter<FROM, TO> {
    * @return
```
Comment (mchades) [inline catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/converter/JdbcTypeConverter.java:24]: Please full the comment

File: catalogs/catalog-jdbc-postgresql/src/main/java/com/datastrato/gravitino/catalog/postgresql/converter/PostgreSqlTypeConverter.java [hunk 5]
```diff
@@ -7,87 +7,83 @@
...
+  public Type toGravitinoType(JdbcTypeBean typeBean) {
     // TODO #947 Complex types are not considered for support in this issue, which will bring more
     // testing needs
-    switch (type.getDataType().toLowerCase()) {
-      case "boolean":
+    switch (typeBean.getTypeName().toLowerCase()) {
+      case "bool":
         return Types.BooleanType.get();
-      case "smallint":
+      case "int2":
```
Comment (mchades) [inline catalogs/catalog-jdbc-postgresql/src/main/java/com/datastrato/gravitino/catalog/postgresql/converter/PostgreSqlTypeConverter.java:19]: Can you use a static parameter for the const string?

Comment (FANNG1) [reply catalogs/catalog-jdbc-postgresql/src/main/java/com/datastrato/gravitino/catalog/postgresql/converter/PostgreSqlTypeConverter.java:19]: +1

File: catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java [hunk 6]
```diff
@@ -231,4 +227,26 @@ protected Connection getConnection(String catalog) throws SQLException {
     connection.setCatalog(catalog);
     return connection;
   }
+
+  protected JdbcTable.Builder createCommonTableBuilder(ResultSet table) throws SQLException {
```
Comment (FANNG1) [inline catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java:231]: createJdbcTableBuilder?

File: catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlTableOperations.java [hunk 7]
```diff
@@ -58,39 +56,24 @@ public JdbcTable load(String databaseName, String tableName) throws NoSuchTableE
...
-            new JdbcColumn.Builder()
-                .withName(columnName)
-                .withType(typeConverter.toGravitinoType(typeBean))
-                .withNullable(columns.getBoolean("NULLABLE"))
-                .withComment(StringUtils.isEmpty(colComment) ? null : colComment)
-                // TODO #1531 will add default value.
-                //                .withDefaultValue(columns.getString("COLUMN_DEF"))
-                .withProperties(Collections.emptyList())
+            createCommonColumnBuilder(columns)
+                .withAutoIncrement("YES".equalsIgnoreCase(columns.getString("IS_AUTOINCREMENT")))
```
Comment (FANNG1) [inline catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlTableOperations.java:61]: this should in common process logic

File: catalogs/catalog-jdbc-postgresql/src/main/java/com/datastrato/gravitino/catalog/postgresql/operation/PostgreSqlTableOperations.java [hunk 8]
```diff
@@ -84,139 +55,30 @@ public void initialize(
   @Override
   public JdbcTable load(String schema, String tableName) throws NoSuchTableException {
```
Comment (FANNG1) [inline catalogs/catalog-jdbc-postgresql/src/main/java/com/datastrato/gravitino/catalog/postgresql/operation/PostgreSqlTableOperations.java:56]: could you put the common logic in JdbcTableOperations, leave the specified logic in XXTableOperations?  since Mysql obtain most field from Jdbc drivers too.

File: catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlTableOperations.java (hunk 7 above)
Comment (FANNG1) [reply catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlTableOperations.java:61]: I couldn't understand, please describe more details.

File: catalogs/catalog-jdbc-postgresql/src/main/java/com/datastrato/gravitino/catalog/postgresql/operation/PostgreSqlTableOperations.java (hunk 8 above)
Comment (FANNG1) [reply catalogs/catalog-jdbc-postgresql/src/main/java/com/datastrato/gravitino/catalog/postgresql/operation/PostgreSqlTableOperations.java:56]: how about put `createCommonTableBuilder ` in  JdbcCatalogOperations?

File: catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java [hunk 9]
```diff
@@ -106,32 +102,31 @@ public List<String> listTables(String databaseName) throws NoSuchSchemaException
...
-          result.add(extractJdbcColumnFromResultSet(column));
-        }
-        jdbcColumns = result.toArray(new JdbcColumn[0]);
+
+      // 2.Get column information
+      List<JdbcColumn> jdbcColumns = new ArrayList<>();
+      ResultSet columns = getColumns(connection, databaseName, tableName);
+      while (columns.next()) {
+        JdbcColumn.Builder columnBuilder = createJdbcColumnBuilder(columns);
+        extractJdbcColumnFromResultSet(columns, columnBuilder);
```
Comment (FANNG1) [inline catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java:117]: `extractJdbcColumnFromResultSet` `extractJdbcTableFromResultSet` `extractJdbcTable` are too general names. I couldn't get much information from the name. may be we could use `getAutoIncreamentInfo()` `getIndexInfo()` to  get responding context.  `createJdbcXXBuilder` suggest to rename to `getBasicJdbcxxInfo`. the code maybe like
```java
        JdbcColumn.Builder columnBuilder = getBasicJdbcColumnInfo(columns);
        Bool autoIncrement = getAutoIncreamentInfo(columns);
        collumnBuilder.withAutoIncrement(autoIncrement)
```

File: catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java [hunk 10]
```diff
@@ -106,32 +102,31 @@ public List<String> listTables(String databaseName) throws NoSuchSchemaException
...
-          .withColumns(jdbcColumns)
-          .withComment(comment)
-          .withProperties(properties)
-          .withAuditInfo(AuditInfo.EMPTY)
-          .build();
-    } catch (final SQLException se) {
-      throw this.exceptionMapper.toGravitinoException(se);
+
+      JdbcTable.Builder jdbcTableBuilder = createJdbcTableBuilder(table);
+      extractJdbcTableFromResultSet(table, jdbcTableBuilder);
```
Comment (FANNG1) [inline catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java:122]: what's the usage of extractJdbcTableFromResultSet?

File: catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/converter/JdbcTypeConverter.java [hunk 11]
```diff
@@ -5,22 +5,91 @@
...
+import java.util.Objects;
 
-public abstract class JdbcTypeConverter<FROM, TO> {
+/** @param <TO> Implement the corresponding JDBC data type to be converted */
+public abstract class JdbcTypeConverter<TO> {
 
   /**
    * Convert from Gravitino type to JDBC type
    *
    * @param type
```
Comment (FANNG1) [inline catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/converter/JdbcTypeConverter.java:16]: add description about `type`

File: catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/converter/JdbcTypeConverter.java [hunk 12]
```diff
@@ -5,22 +5,91 @@
...
 
 import com.datastrato.gravitino.rel.types.Type;
+import java.util.Objects;
 
-public abstract class JdbcTypeConverter<FROM, TO> {
+/** @param <TO> Implement the corresponding JDBC data type to be converted */
+public abstract class JdbcTypeConverter<TO> {
 
   /**
    * Convert from Gravitino type to JDBC type
```
Comment (FANNG1) [inline catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/converter/JdbcTypeConverter.java:14]: the description about `toGravitinoType` and `fromGravitinoType` are wrong

File: catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java [hunk 13]
```diff
@@ -108,35 +104,36 @@ public List<String> listTables(String databaseName) throws NoSuchSchemaException
...
-          .withProperties(properties)
-          .withAuditInfo(AuditInfo.EMPTY)
-          .build();
-    } catch (final SQLException se) {
-      throw this.exceptionMapper.toGravitinoException(se);
+
+      JdbcTable.Builder jdbcTableBuilder = createJdbcTableBuilder(table);
+
+      // 3.The underlying implementation class customizes the extension of table information.
+      extractJdbcTable(connection, tableName, jdbcTableBuilder);
```
Comment (FANNG1) [inline catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java:127]: It's more clear to split `extractJdbcTable` into 3 functinons:
1. getTableProperties()
2. getIndexs()
3. fill up the meta couldn't find from driver, I couldn't figure out a property name.

File: catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java [hunk 14]
```diff
@@ -108,35 +104,36 @@ public List<String> listTables(String databaseName) throws NoSuchSchemaException
...
+
+      return jdbcTableBuilder.withColumns(jdbcColumns.toArray(new JdbcColumn[0])).build();
+    } catch (SQLException e) {
+      throw exceptionMapper.toGravitinoException(e);
     }
   }
 
+  protected abstract boolean getAutoIncrementInfo(ResultSet columns) throws SQLException;
```
Comment (FANNG1) [inline catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java:135]: provide a default function for `getAutoIncrementInfo`, some other JDBC drivers may not care getAutoIncrementInfo, they don't need to implement it.

File: catalogs/catalog-jdbc-postgresql/src/main/java/com/datastrato/gravitino/catalog/postgresql/operation/PostgreSqlTableOperations.java [hunk 15]
```diff
@@ -570,4 +395,22 @@ protected Connection getConnection(String schema) throws SQLException {
...
+    // Directly using `connection.getSchema()` cannot retrieve the corresponding schema, so here
+    // `schema` is used.
+    return metaData.getTables(database, schema, tableName, null);
+  }
+
+  @Override
+  protected ResultSet getColumns(Connection connection, String schema, String tableName)
+      throws SQLException {
+    DatabaseMetaData metaData = connection.getMetaData();
+    // Directly using `connection.getSchema()` cannot retrieve the corresponding schema, so here
```
Comment (FANNG1) [inline catalogs/catalog-jdbc-postgresql/src/main/java/com/datastrato/gravitino/catalog/postgresql/operation/PostgreSqlTableOperations.java:412]: correct the description

File: catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java (hunk 14 above)
Comment (FANNG1) [reply catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java]: which implement is more general for other JDBC? If there's no answer, use return false as default?

File: catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java [hunk 16]
```diff
@@ -108,35 +104,40 @@ public List<String> listTables(String databaseName) throws NoSuchSchemaException
...
-      Map<String, String> properties;
-      try (ResultSet table = getTable(connection, tableName)) {
-        if (!table.next()) {
-          throw new NoSuchTableException("Table " + tableName + " does not exist.");
-        }
-        comment = table.getString("REMARKS");
-        properties = extractPropertiesFromResultSet(table);
+      // 1.Get table information
+      ResultSet table = getTable(connection, databaseName, tableName);
+      if (!table.next() || !tableName.equals(table.getString("TABLE_NAME"))) {
```
Comment (yuqi1129) [inline catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java:109]: I think we should think twice about cases where tableName is case insensitive.

File: catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlTableOperations.java [hunk 17]
```diff
@@ -309,10 +194,68 @@ public static void appendIndexesSql(Index[] indexes, StringBuilder sqlBuilder) {
...
+  protected boolean getAutoIncrementInfo(ResultSet resultSet) throws SQLException {
+    return "YES".equalsIgnoreCase(resultSet.getString("IS_AUTOINCREMENT"));
+  }
+
+  @Override
+  protected void extractJdbcTable(
+      Connection connection, String tableName, JdbcTable.Builder tableBuilder) throws SQLException {
+    Map<String, String> tableProperties = loadTablePropertiesFromSql(connection, tableName);
+    tableBuilder.withProperties(
+        Collections.unmodifiableMap(
```
Comment (yuqi1129) [inline catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlTableOperations.java]: You can simply the code here:
```
ImmutableMap.builder()
            .put()
            .build()
```

File: catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java [hunk 18]
```diff
@@ -234,4 +246,25 @@ protected Connection getConnection(String catalog) throws SQLException {
...
+  protected JdbcColumn.Builder getBasicJdbcColumnInfo(ResultSet column) throws SQLException {
+    JdbcTypeConverter.JdbcTypeBean typeBean =
+        new JdbcTypeConverter.JdbcTypeBean(column.getString("TYPE_NAME"));
+    typeBean.setColumnSize(column.getString("COLUMN_SIZE"));
+    typeBean.setScale(column.getString("DECIMAL_DIGITS"));
+    String colComment = column.getString("REMARKS");
+    return new JdbcColumn.Builder()
+        .withName(column.getString("COLUMN_NAME"))
+        .withType(typeConverter.toGravitinoType(typeBean))
+        .withComment(StringUtils.isEmpty(colComment) ? null : colComment)
```
Comment (FANNG1) [inline catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java:266]: why not use `.withComment(colComment)`?

File: catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/converter/MysqlTypeConverter.java [hunk 19]
```diff
@@ -7,14 +7,12 @@
 import com.datastrato.gravitino.catalog.jdbc.converter.JdbcTypeConverter;
 import com.datastrato.gravitino.rel.types.Type;
 import com.datastrato.gravitino.rel.types.Types;
-import java.util.Objects;
 
 /** Type converter for MySQL. */
-public class MysqlTypeConverter
-    extends JdbcTypeConverter<MysqlTypeConverter.MysqlTypeBean, String> {
+public class MysqlTypeConverter extends JdbcTypeConverter<String> {
```
Comment (FANNG1) [inline catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/converter/MysqlTypeConverter.java:12]: could you use static constant like in PGTypeConverter?

File: catalogs/catalog-jdbc-postgresql/src/main/java/com/datastrato/gravitino/catalog/postgresql/converter/PostgreSqlTypeConverter.java [hunk 20]
```diff
@@ -7,89 +7,103 @@
 import com.datastrato.gravitino.catalog.jdbc.converter.JdbcTypeConverter;
 import com.datastrato.gravitino.rel.types.Type;
 import com.datastrato.gravitino.rel.types.Types;
-import java.util.List;
-import net.sf.jsqlparser.statement.create.table.ColDataType;
 
-public class PostgreSqlTypeConverter extends JdbcTypeConverter<ColDataType, String> {
+public class PostgreSqlTypeConverter extends JdbcTypeConverter<String> {
```
Comment (FANNG1) [inline catalogs/catalog-jdbc-postgresql/src/main/java/com/datastrato/gravitino/catalog/postgresql/converter/PostgreSqlTypeConverter.java:11]: add  UT for xxTypeConverter? or do it in another PR

File: catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java [hunk 21]
```diff
@@ -108,35 +104,50 @@ public List<String> listTables(String databaseName) throws NoSuchSchemaException
...
+
+      // 5.Leave the information to the bottom layer to append the table
+      correctJdbcTableFields(connection, tableName, jdbcTableBuilder);
+      return jdbcTableBuilder.build();
+    } catch (SQLException e) {
+      throw exceptionMapper.toGravitinoException(e);
     }
   }
 
+  protected Map<String, String> getProperties(Connection connection, String tableName)
```
Comment (FANNG1) [inline catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java:142]: rename to getTableProperties?   
please add description about will add extra properties like comment and will be filtered in xx

File: catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java [hunk 22]
```diff
@@ -190,27 +201,28 @@ protected ResultSet getTables(Connection connection) throws SQLException {
...
-    return metaData.getColumns(databaseName, databaseName, tableName, null);
+    return metaData.getColumns(connection.getCatalog(), connection.getSchema(), tableName, null);
   }
 
-  /**
-   * @param table The result set of the table
-   * @return The properties extracted from the result set
-   */
-  protected abstract Map<String, String> extractPropertiesFromResultSet(ResultSet table);
+  protected void correctJdbcTableFields(
```
Comment (FANNG1) [inline catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java:216]: please add the description why add `correctJdbcTableFields` function

File: catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java [hunk 23]
```diff
@@ -108,35 +104,58 @@ public List<String> listTables(String databaseName) throws NoSuchSchemaException
...
+    } catch (SQLException e) {
+      throw exceptionMapper.toGravitinoException(e);
     }
   }
 
+  /**
+   * @param connection jdbc connection
+   * @param tableName table name
+   * @return Returns all table attribute values, which may contain attributes that are not needed in
+   *     Gravitino. The JdbcCatalogOperationsloadTable method will filter out unnecessary
```
Comment (FANNG1) [inline catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java:146]: JdbcCatalogOperationsloadTable -> loadTable in JdbcCatalogOperations?

**Comment by jerryshao** [comment]
Seems this PR also adds index support for PG, am I right?

File: catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java [hunk 24]
```diff
@@ -190,27 +209,37 @@ protected ResultSet getTables(Connection connection) throws SQLException {
...
     final DatabaseMetaData metaData = connection.getMetaData();
-    String databaseName = connection.getSchema();
-    return metaData.getColumns(databaseName, databaseName, tableName, null);
+    return metaData.getColumns(connection.getCatalog(), connection.getSchema(), tableName, null);
   }
 
   /**
-   * @param table The result set of the table
-   * @return The properties extracted from the result set
+   * Different data sources have their own specific content that needs to be processed, such as the
```
Comment (FANNG1) [inline catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java:225]: correct table information from the JDBC driver, because not all information could be retrieved from the JDBC driver, like the table comment in MySQL of the 5.7 version.

File: catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java [hunk 25]
```diff
@@ -190,27 +209,37 @@ protected ResultSet getTables(Connection connection) throws SQLException {
...
+   * @param connection jdbc connection
+   * @param tableName table name
+   * @param jdbcTableBuilder The builder of the table to be returned
+   * @throws SQLException
    */
-  protected abstract Map<String, String> extractPropertiesFromResultSet(ResultSet table);
+  protected void correctJdbcTableFields(
+      Connection connection, String tableName, JdbcTable.Builder jdbcTableBuilder)
+      throws SQLException {
+    // nothing to do
```
Comment (jerryshao) [inline catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java:237]: What is the purpose of this method?

File: catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java [hunk 26]
```diff
@@ -234,4 +263,24 @@ protected Connection getConnection(String catalog) throws SQLException {
...
+    JdbcTypeConverter.JdbcTypeBean typeBean =
+        new JdbcTypeConverter.JdbcTypeBean(column.getString("TYPE_NAME"));
+    typeBean.setColumnSize(column.getString("COLUMN_SIZE"));
+    typeBean.setScale(column.getString("DECIMAL_DIGITS"));
+    return new JdbcColumn.Builder()
+        .withName(column.getString("COLUMN_NAME"))
+        .withType(typeConverter.toGravitinoType(typeBean))
+        .withComment(column.getString("REMARKS"))
+        .withNullable(column.getBoolean("NULLABLE"))
+        .withProperties(Collections.emptyList());
```
Comment (jerryshao) [inline catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java:285]: Is this `withProperties` still useful?

**Comment by FANNG1** [comment]
LGTM

**Review by jerryshao — APPROVED** [review]
